crates/cargo-codspeed/src/measurement_mode.rs

```diff
@@ -1,20 +1,20 @@
 use clap::ValueEnum;
 use serde::Serialize;
-use std::env;
+use std::path::Path;
 
-#[derive(Debug, Clone, ValueEnum, Serialize, PartialEq, Eq)]
+#[derive(Debug, Clone, ValueEnum, Serialize, PartialEq, Eq, Default)]
 #[serde(rename_all = "lowercase")]
 pub enum MeasurementMode {
-    Walltime,
+    #[default]
     Instrumentation,
+    Walltime,
 }
 
-impl Default for MeasurementMode {
-    fn default() -> Self {
-        if env::var("CODSPEED_ENV").is_ok() {
-            MeasurementMode::Instrumentation
-        } else {
-            MeasurementMode::Walltime
+impl AsRef<Path> for MeasurementMode {
+    fn as_ref(&self) -> &Path {
+        match self {
+            MeasurementMode::Walltime => Path::new("walltime"),
+            MeasurementMode::Instrumentation => Path::new("instrumentation"),
         }
     }
 }
```

Comment:
I think we should have a more generic serialization instead of this specific AsRef<Path>